### ISSUE TITLE
Can't add Open Tab to a Container

### ISSUE THREAD

**Issue by aplatypus** [mozilla/testpilot-containers#317]
G'day ...

I think you need to consider the method for **Add**-ing a URL to a container. I am starting to set-up  a container for my work wiki.  I have a lot of tabs/url-s for various wiki pages.  

I would like to be able  to drag the TAB to the Container named "wiki" -- I found that impossible.

You definitely need to look at a simple way to do that -- I suggest something like the existing Bookmarks CTRL/SHIFT-B

Thank you,

   aplatypus

**Comment by smichel17** [comment]
A way to move a tab into/between containers would be very useful to me as well. The first thing I did on install was check whether I could move my existing tabs into containers. I expected that there would be a `Move to container` submenu when right clicking on a tab.

**Comment by smichel17** [comment]
Related: #311

**Comment by a2sheppy** [comment]
Yes please. It's far too easy to wind up with a tab that's not in a container and needs to be moved into one. This is extra important when you first starting using containers, and have lots of already-open tabs that need to be categorized.

**Comment by groovecoder** [comment]
@a2sheppy thanks for the feedback. Can you add your comment to #311 too? We're starting to do the product/UX work on that one.

**Comment by jonathanKingston** [comment]
@aplatypus have you tried the new containers assignment feature.
Thinking about this some more, would it make more sense at setup time?

1. User installs containers
2. User is asked to categorise existing tabs
3. Opening those websites again assigns to a the container mentioned.
4. No further option is exposed to convert one tab to another.

Either way I think this makes sense in some form of management screen of assignments making it clear they should remove existing cookies

**Comment by grahamperrin** [comment]
> … containers assignment …

I can't find the feature.

**Comment by jonathanKingston** [comment]
Hey @grahamperrin,

Unfortunately moving from one container to another is a privacy issue which we really don't want to open up. Whilst this makes tab groups style usage an issue it would reduce the real reason why we are doing containers.

We are however all ears to ideas on how to solve this... without just moving across the container boundary. Moving from one container to another leaks cookie and tracking info from the sites in one container to another.

> I can't find the feature.

We are working on making this more discoverable, thank you for hunting it down along with the commits etc!

Anyway thanks for your interest, I want to keep this open as maybe there is a path through here.

@aplatypus you can actually drag the (i) icon of the tab into a different tab. This copies the URL into a different container.

There probably is rationale on setup of a container to allow this behaviour too, however building it into common work flows isn't something I think we should support.

**Comment by smichel17** [comment]
@jonathanKingston in https://github.com/mozilla/testpilot-containers/issues/336#issuecomment-300666229 you said,

> After the initial hump of creating a container, I have rarely found I want to move containers.

For me it's mostly use case 2 in https://github.com/mozilla/testpilot-containers/issues/428#issuecomment-291956953. When I'm browsing and I think of something I want to do/say using a different account, it's fairly cumbersome to click on the address bar, copy the url, open a new tab in the desired container, paste the url, hit enter, and then go find my place in the page. Even more so if I start to type my response before realizing I want to change accounts; that adds another several clicks and keystrokes switching between tabs.

Most of the time I do not actually want to convert the current tab, because I also want to continue using it afterwards; what I really want is a *quick and easy* shortcut that **duplicates** the current \_URL\_ into a new container. *Maybe* it also scrolls down to the current position (that's a pretty big ask and I wouldn't expect it in test pilot). Bonus points if it strips identifying info like [clean links](https://github.com/diegocr/CleanLinks) ~~does~~ did (also out of scope for now).

I will agree that a significant portion of the pain here is how clunky it is to choose which container you want to open a new tab in (unreliably hover, no keyboard shortcuts, searching through 10 different containers to find the one I want). It looks like the extension you made recently might help with that, although I anticipate it won't completely resolve the problem. (I'd enjoy being proved wrong, though!)

**Comment by a2sheppy** [comment]
The interesting thing here is that I don’t know how many people will think of this as a privacy feature in the real world. To me, and probably to a lot of others, this is a tab organization feature. I want to group thing by context so I can more easily context-switch between work and play, for example. I don’t really worry much about stuff leaking over because I’m careful about it in other ways.

> On May 10, 2017, at 9:28 PM, Jonathan Kingston <notifications@github.com> wrote:
> 
> Unfortunately moving from one container to another is a privacy issue which we really don't want to open up. Whilst this makes tab groups style usage an issue it would reduce the real reason why we are doing containers.
> 
> 


Eric Shepherd
sheppy@sheppyware.net

**Comment by jonathanKingston** [comment]
@a2sheppy reimplementing this extension to not be about containers wouldn't be difficult. Tab management extensions will come in numbers does it really make sense to try and make this solve this use-case?

The only way I can see us shipping this is having a "Yes I understand, I just want tab highlighting" to enable these conversion features.

The fact is extensions already can and have solved this use case.

I would much rather solve:
- Manage tabs into the new containers when:
  - Enabling containers
  - Creating a new container
- Having a safer default tab which is isolated and allows you to move it into a container **only once**
- Providing easier tools/methods for users to pick the right container when visiting a tab to reduce the "ah crap wrong container" feeling.

**Comment by jplatte** [comment]
@jonathanKingston I'd be perfectly fine with reloading the page when it is moved into a container. Maybe it should also be "reload in container" instead of "move to container". That should resolve the privacy issue, right?

**Comment by jonathanKingston** [comment]
@jplatte as I mentioned here https://github.com/mozilla/testpilot-containers/issues/311#issuecomment-306459964 I think users should just install contextPlus as this reduces the privacy benefit completely.

> It reloads the page in a different container. Is the concern here that reloading a page in a container could allow the server to detect that it is the same browser because the URL might contain identifying information

Yup that is exactly the issue, moving containers even just the URL can identify the user across a container. Sure the risk is usually low however it is possible. Filtering parameters would require some form of add blocking tech and even that will never be perfect.

I think this is actually a duplicate of that bug anyway unless I'm not understanding the difference.

**Comment by jplatte** [comment]
Alright, I'll re-recommend Context Plus with a [link](https://addons.mozilla.org/nn-no/firefox/addon/context-plus/) then. And I agree that this issue can be closed as a dup of #311.

**Comment by jonathanKingston** [comment]
I think this is a duplicate of #311 and I also just closed that. Please reopen if you think there is a distinction.

**Comment by groovecoder** [comment]
I think @jonathanKingston meant to close this when he marked it a duplicate of #311. (The resolution there was to use [Context Plus](https://addons.mozilla.org/nn-no/firefox/addon/context-plus/?src=search) for this.)
